test/test_connectionpool.py

```diff
@@ -49,6 +49,7 @@ def test_same_host(self):
             ('http://google.com/', 'http://google.com:80/abracadabra'),
             ('https://google.com:443/', 'https://google.com/abracadabra'),
             ('https://google.com/', 'https://google.com:443/abracadabra'),
+            ( 'http://[2607:f8b0:4005:805::200e%25eth0]/', 'http://[2607:f8b0:4005:805::200e%eth0]/'),
```

Comment:
Can we extend this test a fair bit? Let's add a few other IPv6 address formats to confirm that they are all processed sensibly.